server/src/internalClusterTest/java/org/opensearch/indices/replication/SearchReplicaRestoreIT.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.indices.replication;
+
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.indices.replication.common.ReplicationType;
+import org.opensearch.snapshots.AbstractSnapshotIntegTestCase;
+import org.opensearch.snapshots.SnapshotRestoreException;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.List;
+
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_NUMBER_OF_SEARCH_REPLICAS;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertHitCount;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class SearchReplicaRestoreIT extends AbstractSnapshotIntegTestCase {
+
+    private static final String INDEX_NAME = "test-idx-1";
+    private static final String RESTORED_INDEX_NAME = INDEX_NAME + "-restored";
+    private static final String REPOSITORY_NAME = "test-repo";
+    private static final String SNAPSHOT_NAME = "test-snapshot";
+    private static final String FS_REPOSITORY_TYPE = "fs";
+    private static final int DOC_COUNT = 10;
+
+    @Override
+    protected Settings featureFlagSettings() {
+        return Settings.builder().put(super.featureFlagSettings()).put(FeatureFlags.READER_WRITER_SPLIT_EXPERIMENTAL, true).build();
+    }
+
+    public void testSearchReplicaRestore_WhenSnapshotOnDocRep_RestoreOnDocRepWithSearchReplica() throws Exception {
+        bootstrapIndexWithOutSearchReplicas(ReplicationType.DOCUMENT);
+        createRepoAndSnapshot(REPOSITORY_NAME, FS_REPOSITORY_TYPE, SNAPSHOT_NAME, INDEX_NAME);
+
+        SnapshotRestoreException exception = expectThrows(
+            SnapshotRestoreException.class,
+            () -> restoreSnapshot(
+                REPOSITORY_NAME,
+                SNAPSHOT_NAME,
+                INDEX_NAME,
+                RESTORED_INDEX_NAME,
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.DOCUMENT)
+                    .put(SETTING_NUMBER_OF_SEARCH_REPLICAS, 1)
+                    .build()
+            )
+        );
+        assertTrue(exception.getMessage().contains(getSnapshotExceptionMessage(ReplicationType.DOCUMENT, ReplicationType.DOCUMENT)));
+    }
+
+    public void testSearchReplicaRestore_WhenSnapshotOnDocRep_RestoreOnSegRep() throws Exception {
```

Comment:
We don't need this case here, this is covered by existing segrep restore tests.  Only the cases where search replicas are involved

Reply:
Sure